src/direction.rs

```diff
@@ -46,8 +46,17 @@ impl Direction {
     /// Move a point in this direction by the given amount.
     #[inline]
     #[must_use]
-    pub fn move_point(&self, point: IVec2, by: i32) -> IVec2 {
-        point + self.unit() * by
```

Comment:
Does this not work? Or maybe it's not as good because of too many conversions?
```rust
(point.as_ivec2() + self.unit() * by as i32).as_uvec2()
```

Reply:
Yes, much cleaner!